test/coverage/utils/ExpectedTest.cpp

```diff
@@ -12,8 +12,50 @@
 #include <gtest/gtest.h>
 #include <up-cpp/utils/Expected.h>
 
+#include <cmath>
+
 namespace {
 
+using uprotocol::utils::BadExpectedAccess;
+using uprotocol::utils::Expected;
+using uprotocol::utils::Unexpected;
+
+enum class parse_error { invalid_input, overflow };
+
+struct CompositeExpect {
+	double x;
+	double y;
+};
+
+auto parse_number(std::string_view str) -> Expected<double, parse_error> {
+	const char* begin = str.data();
+	char* end;
+	double retval = std::strtod(begin, &end);
+
+	if (begin == end)
+		return Unexpected(parse_error::invalid_input);
+	else if (std::isinf(retval))
+		return Unexpected(parse_error::overflow);
+
+	str.remove_prefix(end - begin);
+	return retval;
+}
+
+auto parse_number_with_composite(std::string_view str)
+    -> Expected<CompositeExpect, parse_error> {
+	const char* begin = str.data();
+	char* end;
+	double retval = std::strtod(begin, &end);
+
+	if (begin == end)
+		return Unexpected(parse_error::invalid_input);
+	else if (std::isinf(retval))
+		return Unexpected(parse_error::overflow);
+
+	str.remove_prefix(end - begin);
+	return CompositeExpect{retval, -retval};
+}
```

Comment:
Could `Expected` be tested more directly in the test cases? Most of the code here isn't really relevant to the operation of `Expected` and obscures what is actually being tested.

Reply:
The function pattern started because it was in the example8 on the cppreference page for expected. I think it highlights the common use case, which is to check for failures in functions. The functions could be done away with, but that seems less precise to me. I would think, if anything, there should be additional test cases for assignment with expected, not lieu of the functions.